src/server/raft_client.rs

```diff
@@ -64,7 +65,24 @@ impl Conn {
         let client2 = client1.clone();
 
         let (tx, rx) = batch::unbounded::<RaftMessage>(RAFT_MSG_NOTIFY_SIZE);
-        let rx = batch::BatchReceiver::new(rx, RAFT_MSG_MAX_BATCH_SIZE, Vec::new, |v, e| v.push(e));
+        struct RaftMsgCollector(usize);
+        impl BatchCollector<Vec<RaftMessage>, RaftMessage> for RaftMsgCollector {
+            fn collect(&mut self, v: &mut Vec<RaftMessage>, e: RaftMessage) -> Option<RaftMessage> {
+                let msg_size = e.compute_size() as usize;
+                if self.0 > 0 && self.0 + msg_size >= MAX_GRPC_SEND_MSG_LEN as usize {
```

Comment:
What's the point of checking `self.0 > 0`?

Reply:
Without the check, an empty vector could be spilled out if e.len() > MAX_GRPC_SEND_MSG_LEN.